File: code/test_suites/cpp/FunctionBuildTest/tests/test_decimal_infinite.cpp
#include <gtest/gtest.h>
#include "../src/function_decimal_infinite.cpp"

class TestInfiniteFunction : public ::testing::Test {
protected:
    InfiniteCheck infinite_check_func;
};

TEST_F(TestInfiniteFunction, TestPositiveInfinity) {
    std::string result = infinite_check_func.check_discount_for_large_order("Infinity");
    EXPECT_EQ(result, "True");
}

TEST_F(TestInfiniteFunction, TestNegativeInfinity) {
    std::string result = infinite_check_func.check_discount_for_large_order("-Infinity");
    EXPECT_EQ(result, "True");
}

TEST_F(TestInfiniteFunction, TestZero) {
    std::string result = infinite_check_func.check_discount_for_large_order("0.0");
    EXPECT_EQ(result, "False");
}

TEST_F(TestInfiniteFunction, TestFiniteNumber) {
    std::string result = infinite_check_func.check_discount_for_large_order("10.5");
    EXPECT_EQ(result, "False");
}

TEST_F(TestInfiniteFunction, TestNaN) {
    std::string result = infinite_check_func.check_discount_for_large_order("NaN");
    EXPECT_EQ(result, "False");
}

int main(int argc, char **argv) {
    ::testing::InitGoogleTest(&argc, argv);
    return RUN_ALL_TESTS();
}

File: manual/data/python/tests/test_decimal_infinite.py
import unittest
from function_decimal_infinite import InfiniteCheck

class TestInfiniteFunction(unittest.TestCase):
    
    def setUp(self):
        self.infinite_check_func = InfiniteCheck()

    def test_positive_infinity(self):
        value = "Infinity"
        self.assertEqual(self.infinite_check_func.check_discount_for_large_order(value), "True")

    def test_negative_infinity(self):
        value = "-Infinity"
        self.assertEqual(self.infinite_check_func.check_discount_for_large_order(value), "True")

    def test_zero(self):
        value = "0.0"
        self.assertEqual(self.infinite_check_func.check_discount_for_large_order(value), "False")

    def test_finite_number(self):
        value = "10.5"
        self.assertEqual(self.infinite_check_func.check_discount_for_large_order(value), "False")

    def test_nan(self):
        value = "NaN"
        self.assertEqual(self.infinite_check_func.check_discount_for_large_order(value), "False")

if __name__ == '__main__':
    unittest.main()
